src/Input.js

```diff
@@ -0,0 +1,61 @@
+import {
+  ERROR_MESSAGE_CAR_NAME_DUPLICATION,
+  ERROR_MESSAGE_CAR_NAME_INVALID,
+  ERROR_MESSAGE_USER_INPUT,
+  INPUT_MESSAGE_CAR_NAMES,
+  INPUT_MESSAGE_TRY_COUNT,
+} from './lib/constants.js';
+import {
+  checkArrayAllUnique,
+  readLineAsync,
+  splitIntoArray,
+  validatePositiveInteger,
+} from './lib/utils.js';
+
+class Input {
+  static #CAR_NAME_REGEXP = /^\w{1,5}$/;
+
+  static #SEPARATOR = ',';
+
+  #rawCars;
+
+  #rawTryCount;
+
+  constructor() {
+    this.#rawCars = '';
+    this.#rawTryCount = '';
+  }
+
+  async getUserInput() {
+    try {
+      this.#rawCars = await readLineAsync(INPUT_MESSAGE_CAR_NAMES);
+      this.#rawTryCount = await readLineAsync(INPUT_MESSAGE_TRY_COUNT);
+    } catch (e) {
+      throw new Error(ERROR_MESSAGE_USER_INPUT);
+    }
+  }
+
+  parseCars() {
+    const carArray = splitIntoArray(this.#rawCars, Input.#SEPARATOR);
+    Input.#validateCarArray(carArray);
```

Comment:
```suggestion
    this.#validateCarArray(carArray);
```
이렇게 작성하지 않으신 이유가 있으신가요?

Reply:
처음에는 `validateCarArray`내부에서 인스턴스 프로퍼티에 접근하지 않기 때문에 정적 메서드로 선언을 했습니다.

그런데 다시 생각해보니, 인스턴스의 프로퍼티로 carArray를 등록하고, 이를 `this`로 접근하는 것이 더 적합하다는 생각이 드네요. `validateCarArray`가 결국엔 `Input`인스턴스의 값의 유효성을 체크하는 것이니깐요.

다시 생각해보게되네요 감사합니다.

Reply:
저는 사실 단순하게, `Input`안에 있는 메서드를 `Input` 안에 있는 메서드에서 불러오기 때문에 `this.`을 사용하는 것이 당연하다고 생각했어요 ㅎㅎ 
저두 어떤 경우에 `this`를 사용하고 어떤 경우에 `Input`으로 접근하는 건지 좀 더 공부가 필요하겠어요..!